nucypher/blockchain/eth/registry.py

```diff
@@ -161,6 +161,32 @@ def commit(self, filepath) -> str:
             self.clear()                                  # clear prior sim runs
 
         _ = shutil.copy(self.temp_filepath, filepath)
-        self.temp_filepath = constants.REGISTRY_COMMITED  # just in case
+        self.temp_filepath = constants.REGISTRY_COMMITTED  # just in case
         self.log.info("Wrote temporary registry to filesystem {}".format(filepath))
         return filepath
+
+
+class InMemoryEthereumContractRegistry(EthereumContractRegistry):
+
+    def __init__(self) -> None:
+        super().__init__(registry_filepath=":memory:")
+        self.__registry_data = None  # type: str
+
+    def clear(self):
+        self.__registry_data = list()
+
+    def _swap_registry(self, filepath: str) -> bool:
+        raise NotImplementedError
+
+    def write(self, registry_data: list) -> None:
+        self.__registry_data = json.dumps(registry_data)
+
+    def read(self) -> list:
+        try:
```

Comment:
This needs some comments - when is `TypeError` raised?  Why is it OK if `__registry_data` is `None`.

Reply:
`None` is the default initial state.  Perhaps this can be an empty collection instead.